.gitlab-ci.yml

```
@@ -595,14 +599,31 @@ changelog:
     - test -z "$RELEASE_PLEASE_PR" && echo "No release-please PR found" && exit 0
     - gh pr checkout --force $RELEASE_PLEASE_PR
     - wget --output-document cliff.toml https://raw.githubusercontent.com/mendersoftware/mendertesting/master/utils/cliff.toml
-    - mv CHANGELOG.md.${CI_COMMIT_REF_NAME} CHANGELOG.md
-    - git cliff
-      --unreleased
-      --prepend CHANGELOG.md
-      --github-repo ${GITHUB_REPO_URL}
-      --use-branch-tags
-      --tag $(jq -r '.["."]' .release-please-manifest.json)
-    - git add CHANGELOG.md
+    - if [[ "${THIS_REPO}" == "mender-server-enterprise" ]]; then
+        if [[ $(grep --count "saas" .release-please-manifest.json) -eq 1 ]]; then
+          echo "INFO - running CHANGELOG for saas only";
+          CHANGELOG_SUFFIX="-saas";
+          mv CHANGELOG${CHANGELOG_SUFFIX}.md.${CI_COMMIT_REF_NAME} CHANGELOG${CHANGELOG_SUFFIX}.md;
+          git cliff --unreleased --prepend CHANGELOG${CHANGELOG_SUFFIX}.md --github-repo ${GITHUB_REPO_URL} --use-branch-tags --tag $(jq -r '.["."]' .release-please-manifest.json);
+          git add CHANGELOG${CHANGELOG_SUFFIX}.md;
+        else
+          echo "INFO - stable release - running CHANGELOG for both saas and Enterprise";
+          CHANGELOG_SUFFIX="-saas";
+          mv CHANGELOG${CHANGELOG_SUFFIX}.md.${CI_COMMIT_REF_NAME} CHANGELOG${CHANGELOG_SUFFIX}.md;
+          git cliff --unreleased --prepend CHANGELOG${CHANGELOG_SUFFIX}.md --github-repo ${GITHUB_REPO_URL} --use-branch-tags --tag $(jq -r '.["."]' .release-please-manifest.json);
+          git add CHANGELOG${CHANGELOG_SUFFIX}.md;
+          CHANGELOG_SUFFIX="-enterprise";
+          mv CHANGELOG${CHANGELOG_SUFFIX}.md.${CI_COMMIT_REF_NAME} CHANGELOG${CHANGELOG_SUFFIX}.md;
+          git cliff --unreleased --prepend CHANGELOG${CHANGELOG_SUFFIX}.md --github-repo ${GITHUB_REPO_URL} --use-branch-tags --tag $(jq -r '.["."]' .release-please-manifest.json) --ignore-tags saas;
+          git add CHANGELOG${CHANGELOG_SUFFIX}.md;
+        fi;
+      else
+        echo "INFO - running CHANGELOG for non-enterprise";
+        export CHANGELOG_SUFFIX="";
+        mv CHANGELOG${CHANGELOG_SUFFIX}.md.${CI_COMMIT_REF_NAME} CHANGELOG${CHANGELOG_SUFFIX}.md;
+        git cliff --unreleased --prepend CHANGELOG${CHANGELOG_SUFFIX}.md --github-repo ${GITHUB_REPO_URL} --use-branch-tags --tag $(jq -r '.["."]' .release-please-manifest.json) --ignore-tags saas;
+        git add CHANGELOG${CHANGELOG_SUFFIX}.md;
+      fi;
```

There's also a lot of duplication in the different path. How about refactoring it to something like this?
```suggestion
    - RELEASE_VERSION="$(jq -r '.["."]' .release-please-manifest.json)"
    - case $RELEASE_VERSION in
    - *saas*) CHANGELOG_SUFFIX="-saas";;
    - *) test $CI_PROJECT_NAME = "mender-server-enterprise" && CHANGELOG_SUFFIX="-enterprise";;
    - esac
    - CHANGELOG_FILE="CHANGELOG${CHANGELOG_SUFFIX:-}.md"
    - echo "Generating changelog file ${CHANGELOG_FILE} for release ${RELEASE_VERSION}"
    - mv CHANGELOG${CHANGELOG_SUFFIX}.md.${CI_COMMIT_REF_NAME} CHANGELOG${CHANGELOG_SUFFIX}.md;
    -  git cliff --unreleased --prepend CHANGELOG${CHANGELOG_SUFFIX}.md --github-repo ${GITHUB_REPO_URL} --use-branch-tags --tag $(jq -r '.["."]' .release-please-manifest.json);
    - git add CHANGELOG${CHANGELOG_SUFFIX}.md
```

---

That's why I love when you review my ugly work! Yes, I love this one. 
But, please note that to make CHANGELOG-enterprise works properly, you have to set the `--ignore-tags saas` argument. I'll check it later

---

Additionally, we want to deploy run the `saas` changelog on every change (also for stable tags), while running the enterprise changelog only on stable tags, so the case `*)` cannot work this way

---

That's the best I could do

---

If I understand it correctly we have 4 cases:
 * `mender-server-enterprise/v4.1.0-saas.123`: Updates only `CHANGELOG-saas.md`
 * `mender-server-enterprise/v4.1.0`: Would update both `CHANGELOG-enterprise.md` and `CHANGELOG-saas.md`
 * `mender-server/v4.1.0`: Would update `CHANGELOG.md`
 * `mender-server/v4.1.0-saas.123`: Do not update anything. 
Then, would something like this work?

```suggestion
    - if test $CI_PROJECT_NAME = "mender-server-enterprise"; then
    - case $RELEASE_VERSION in
    - *saas*) CHANGELOG_FILES="CHANGELOG-saas.md";;
    - *) CHANGELOG_FILES="CHANGELOG-enterprise.md CHANGELOG-saas.md"
    - ;;
    - esac
    - else
    - case $RELEASE_VERSION in
    - *saas*) CHANGELOG_FILES="";; # Do nothing
    - *) CHANGELOG_FILES="CHANGELOG.md";;
    - esac
    - fi
    - for CHANGELOG_FILE in $CHANGELOG_FILES; do
    - echo "Generating changelog file ${CHANGELOG_FILE} for release ${RELEASE_VERSION}"
    - mv "${CHANGELOG_FILE}.${CI_COMMIT_REF_NAME}" "$CHANGELOG_FILE";
    - git cliff --unreleased --prepend CHANGELOG${CHANGELOG_SUFFIX}.md --github-repo ${GITHUB_REPO_URL} --use-branch-tags --tag "$RELEASE_VERSION";
    - git add "$CHANGELOG_FILE"
    - done
```
Or maybe it would make sense to put this in it's own script? The formatting is starting to look a bit unreadable :man_shrugging:

---

Yeah, let's write an helper script

---

Done, let's try it now